lib/matplotlib/backends/backend_cairo.py

```diff
@@ -134,6 +134,13 @@ def _fill_and_stroke (self, ctx, fill_c, alpha, alpha_overrides):
 
     @staticmethod
     def convert_path(ctx, path, transform):
+        extents = ctx.clip_extents()
+        bbox = Bbox([[extents[0], extents[1]], [extents[2], extents[3]]])
+        try:
+            path = path.clip_to_bbox (bbox)
```

Comment:
pep8 + commit hooks might be helpful if one doesn't mind losing some history, the autopep8 script does a decent job at fixing these issues.